plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/animators/PuckAnimator.kt

```diff
@@ -65,17 +69,21 @@ internal abstract class PuckAnimator<T>(
     return super.clone()
   }
 
+  fun setUpdateListener(updateListener: ((T) -> Unit)) {
+    if (this.updateListener != updateListener) {
```

Comment:
why not directly set the updateListener? Do we actually call the setUpdateListener multiple times?

Reply:
Yes, via `updateSettings` we call it multiple times. Instead of checking there to do it only once I incapsulated it here.